graphql_compiler/schema/__init__.py

```diff
@@ -447,3 +451,21 @@ def check_for_nondefault_directive_names(directives):
         raise AssertionError(
             u"Unsupported directives found: {}".format(nondefault_directives_found)
         )
+
+
+def compute_schema_fingerprint(schema: GraphQLSchema) -> str:
+    """Compute a fingerprint compactly representing all the data in the given schema.
+
+    The fingerprint is not sensitive to things like type or field order. This function is guaranteed
+    to be robust enough that if two GraphQLSchema have the same fingerprint, then they also
+    represent the same schema.
```

Comment:
```suggestion
    represent the same schema.
    
    Because of internal implementation changes, different versions of this library *may* produce
    different fingerprints for the same schema. Since cross-version fingerprint stability
    is an *explicit non-goal* here, changing a schema's fingerprint will not be considered 
    a breaking change.
```